src/App.tsx

```diff
@@ -1,61 +1,154 @@
+import { useEffect, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import { ToDo } from './types/ToDo';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const todosInfo: ToDo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosInfo);
+  const [user, setUser] = useState('');
+  const [title, setTitle] = useState('');
+  const [hasUser, setHasUser] = useState(false);
+  const [hasTitle, setHasTitle] = useState(false);
```

Comment:
are you sure you need hasUser and hasTitle states? You can just check is there some value inside user and title, isn't it enough?

Reply:
You right. It's useless. Thanks for advice :)